File: week1-c/mario/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

void print_row(int bricks, int hieght);

int main(void)
{
    // prompt the user for input
    int hieght;
    do
    {
        hieght = get_int("What is a hieght of pyramid ? ");
    }
    while (hieght < 1);

    // print the pyramid
    for (int i = 0; i < hieght; i++)
    {
        print_row(i + 1, hieght);
    }
}

void print_row(int bricks, int hieght)
{
    int spaces = hieght - bricks;
    for (int j = 0; j < spaces; j++)
    {
        printf(" ");
    }
    for (int i = 0; i < bricks; i++)
    {
        printf("#");
    }
    printf("\n");
}

File: x/week6-python/sentimental-mario-more/mario.py
from cs50 import get_int


def main():
    # Prompt the user for the height
    while True:
        height = get_int("Height: ")
        if 1 <= height <= 8:
            break

    # Print the pyramid
    for i in range(1, height + 1):
        print_row(i, height)


def print_row(bricks, height):
    spaces = height - bricks

    # Print the right pyramid and the gap
    print(spaces * " ", end="")

    print(bricks * "#", end="  ")

    # Print the left pyramid
    print(bricks * "#")


main()
